### DISCUSSION TITLE
Maximum number of properties in an event?

### DISCUSSION THREAD

**Discussion by theta3** [microsoft/cpp_client_telemetry#1074] [Q&A]
# Is your question about how to consume a C++ library? No
Hi, Azure Data Explorer has [10000 columns limit](https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/schema-entities/columns). Does this library has similar limit on number of properties per event? I can't find it in the code

**Comment by maxgolov** [comment]
Historically the 1DS collector enforces the same restrictions and limitations as legacy SDK collector server-side. You can go to the internal portal that you'd normally use to create a new token. Click on `? Help`, search for `Input constraints`. Maximum _supported_ number is 100 columns. Your mileage may vary - for certain pipeline flows a higher number of properties could work. If you really need a more complex structure, you could collate multiple properties as JSON in a string field, then use Kusto `todynamic()` function to create a JSON object with virtually any number of properties.
